### PR TITLE
Add creation of kueue resources in e2e tests

### PR DESCRIPTION
**Pull request by ChughShilpa**
# Issue link
[RHOAIENG-11047](https://issues.redhat.com/browse/RHOAIENG-11047)

# What changes have been made
Since we are adjusting e2e tests to run with AMD gpu, thus 
- Adding creation of kueue resources as part of test to make it compatible with gpuresources (which is covered in next PR)
- Removing creation of kueue resources  from setup script
- Adjust methods with updated Ray cluster client

# Verification steps
Run test manually 
`go test -timeout 30m -v ./test/e2e -run TestMnistPyTorchAppWrapperCpu`

## Checks
- [ ] I've made sure the tests are passing. 
- Testing Strategy
   - [ ] Unit tests
   - [X] Manual tests
   - [ ] Testing is not required for this change

### CODE REVIEW

File: test/e2e/mnist_rayjob_raycluster_test.go [hunk 1]
```diff
@@ -408,8 +420,42 @@ func getRayDashboardURL(test Test, namespace, rayClusterName string) url.URL {
...
+						Resources: []v1beta1.ResourceQuota{
+							{
+								Name:         corev1.ResourceCPU,
+								NominalQuota: resource.MustParse("8"),
+							},
+							{
+								Name:         corev1.ResourceMemory,
+								NominalQuota: resource.MustParse("12Gi"),
+							},
+							{
```
Comment (astefanutti) [inline test/e2e/mnist_rayjob_raycluster_test.go:449]: Should it be removed if `numberOfGpus` is 0?

File: test/e2e/mnist_rayjob_raycluster_test.go [hunk 2]
```diff
@@ -408,8 +420,42 @@ func getRayDashboardURL(test Test, namespace, rayClusterName string) url.URL {
...
+								Name:         corev1.ResourceName("nvidia.com/gpu"),
+								NominalQuota: resource.MustParse(fmt.Sprint(numberOfGpus)),
+							},
+						},
+					},
+				},
+			},
+		},
 	}
+	clusterQueue := CreateKueueClusterQueue(test, cqSpec)
```
Comment (astefanutti) [inline test/e2e/mnist_rayjob_raycluster_test.go:459]:
```suggestion
	return CreateKueueClusterQueue(test, cqSpec)
```

File: test/e2e/mnist_pytorch_appwrapper_test.go [hunk 3]
```diff
@@ -31,20 +32,26 @@ import (
...
 }
 
 func TestMnistPyTorchAppWrapperGpu(t *testing.T) {
-	runMnistPyTorchAppWrapper(t, "gpu")
+	runMnistPyTorchAppWrapper(t, "gpu", 1)
 }
 
 // Trains the MNIST dataset as a batch Job in an AppWrapper, and asserts successful completion of the training job.
-func runMnistPyTorchAppWrapper(t *testing.T, accelerator string) {
+func runMnistPyTorchAppWrapper(t *testing.T, accelerator string, numberOfGpus int) {
```
Comment (astefanutti) [inline test/e2e/mnist_pytorch_appwrapper_test.go:43]: It's be good we move to an `Option` variadic argument.

File: test/e2e/mnist_rayjob_raycluster_test.go (hunk 1 above)
Comment (ChughShilpa) [reply test/e2e/mnist_rayjob_raycluster_test.go:449]: IIRC specifying the resource name is mandatory even if nominalquota is 0

Comment (ChughShilpa) [reply test/e2e/mnist_rayjob_raycluster_test.go:449]: https://docs.redhat.com/en/documentation/red_hat_openshift_ai_self-managed/2-latest/html/working_with_distributed_workloads/configuring-distributed-workloads_distributed-workloads#configuring-quota-management-for-distributed-workloads_distributed-workloads:~:text=Specify%20the%20resource%20name%20and%20nominalQuota%20in%20the%20flavors.resources%20section%2C%20even%20if%20the%20nominalQuota%20value%20is%200.

Comment (sutaakar) [reply test/e2e/mnist_rayjob_raycluster_test.go:449]: The cleanest solution could be to move creation of ResourceFlavor and ClusterQueue (in the end also RayCluster) into `Test<something>Gpu/Cpu` functions. Each test function would then set properties relevant just to the test function.
That way we would get rid of conditions in test, reducing its reading complexity. Though that would mean a large rewrite.
I am ok with current solution too, just thinking out loud.

Comment (sutaakar) [reply test/e2e/mnist_rayjob_raycluster_test.go:449]: > IIRC specifying the resource name is mandatory even if nominalquota is 0

I think it is (was?) mandatory in RHOAI due to SDK passing GPU values. Not sure if it is still needed.
Anyway in real use cases for finetuning users need GPU.

Comment (ChughShilpa) [reply test/e2e/mnist_rayjob_raycluster_test.go:449]: Created a jira issue for SDK to address this - https://issues.redhat.com/browse/RHOAIENG-13618 
@astefanutti @sutaakar

File: test/e2e/mnist_pytorch_appwrapper_test.go (hunk 3 above)
Comment (ChughShilpa) [reply test/e2e/mnist_pytorch_appwrapper_test.go:43]: Thanks for the suggestion @astefanutti 
Will apply this suggestion in a different PR.

**Review by sutaakar — COMMENTED** [review]
/lgtm

**Comment by ChughShilpa** [comment]
@sutaakar @astefanutti 
Could someone please approve the PR ?

**Comment by sutaakar** [comment]
> We may want to think about a strategy for the creation of cluster resources like ClusterQueues in the context of concurrent execution of tests in the future.

Current strategy is to provide generated name for cluster resources.

**Pull request merged**